daskernetes/core.py

```diff
@@ -294,6 +324,60 @@ def _namespace_default():
     return 'default'
 
 
+class _FakeResponse(object):
+    """ Kubernetes public API expects a response object """
+    def __init__(self, data):
+        self._data = data
+
+    @property
+    def data(self):
+        import json
+        return json.dumps(self._data)
+
+
+def deserialize(x, cls):
+    """ Deserialize object to a Kubernetes object
+
+    Parameters
+    ----------
+    x: object
+        Either a Pod, a dictionary definition, or a filename to a
+        yaml-deserializable definition.
+    cls: type
+        kubernetes.client type like V1PodSpec
+
+    Examples
+    --------
+    >>> deserialize({...}, client.V1PodSpec)
+
+    Returns
+    -------
+    kubernetes.client.V1Pod
+    """
+    if isinstance(x, str):
+        import yaml
+        with open(x) as f:
+            x = yaml.load(f)
+    if isinstance(x, dict):
+        x = client.ApiClient().deserialize(_FakeResponse(x), cls)
```

Comment:
I would love to find a way to make this function work.  @yuvipanda I think that you pointed me to something in kubespawner but after several minutes searching through gitter logs I'm unable to find it.

Reply:
the function below is the hack I used to get around this.  It's entirely specialized for my application.